lib/flipper/notifications/jobs/webhook_notification_job.rb

```diff
@@ -17,7 +17,7 @@ def self.disable_sidekiq_retries
       retry_on Webhooks::NetworkError,
                Webhooks::ServerError,
                attempts: 3,
-               wait:     :exponentially_longer
+               wait:     :polynomially_longer
```

Comment:
I was trying to figure this out by looking at Rails release notes, but didn't find a definitive answer.  Does `polynomially_longer` work for Rails < 7.1?  If not, we may have to consider how to adjust this gem's dependencies so that this gem doesn't break in Rails 7.0 apps.  This is technically an open source gem, so I want to be mindful of potential external users 🤔 .

If `polynomially_longer` works in 7.0, then we should be okay to ship this change as-is.

Reply:
The `:polynomially_longer` key [was introduced on 7.1](https://github.com/rails/rails/blob/v7.1.0/activejob/lib/active_job/exceptions.rb#L167-L186), the [`ActiveJob::Exceptions#determine_delay` implementation on 7.0](https://github.com/rails/rails/blob/v7.0.8.1/activejob/lib/active_job/exceptions.rb#L135-L153) does not contain a reference to it.

We will need to roll this out **after the upgrade** in this case, to maintain the same behaviour. There will be a [deprecation warning](https://github.com/rails/rails/blob/v7.1.0/activejob/lib/active_job/exceptions.rb#L64-L67) in the meantime:

```ruby
ActiveJob.deprecator.warn(<<~MSG.squish)
  `wait: :exponentially_longer` will actually wait polynomially longer and is therefore deprecated.
  Prefer `wait: :polynomially_longer` to avoid confusion and keep the same behavior.
MSG
```

Reply:
We can still rollout the `flipper-notifications` upgrade before that, though.

Reply:
I think we could do this to support both versions:
```suggestion
               wait:     ActiveJob.version < "7.1" ? :exponentially_longer : :polynomially_longer
```
Alternatively we could lock the new gem version to depend on `ActiveJob` >= `7.1`, but any bug fixes would need to be back ported to the previous version as well, so maybe the conditional check is the best approach until Rails 7.0 reaches EOL.